### PR TITLE
feat: add slashsable magnitude fn

### PR DESCRIPTION
**Pull request by shrimalmadhur**

### CODE REVIEW

File: src/contracts/core/AVSDirectory.sol [hunk 1]
```diff
@@ -827,6 +827,42 @@ contract AVSDirectory is
...
     }
 
+    /*
+    * @param operator the operator to get the slashable magnitude for
+    * @param strategy the strategy to get the slashable magnitude for
+    * @param operatorSet the operatorSet to get the slashable magnitude for
+    * 
+    * @return the current slashable magnitude of the given operator, strategy, and operatorSet
+    */
+    function getCurrentSlashableMagnitude(
```
Comment (gpsanant) [inline src/contracts/core/AVSDirectory.sol:837]: `getSlashableMagnitudes`

File: src/contracts/core/AVSDirectory.sol [hunk 2]
```diff
@@ -827,6 +827,42 @@ contract AVSDirectory is
...
+        address operator,
+        IStrategy[] calldata strategy
+    ) public view returns (OperatorSet[] memory, IStrategy[] memory, uint64[] memory) {
+        OperatorSet[] memory operatorSets = operatorSetsMemberOf(operator, 0, inTotalOperatorSets(operator));
+        uint256 maxValues = operatorSets.length * strategy.length;
+        OperatorSet[] memory returnOperatorSets = new OperatorSet[](maxValues);
+        IStrategy[] memory returnStrategies = new IStrategy[](maxValues);
+        uint64[] memory slashableMagnitude = new uint64[](maxValues);
```
Comment (gpsanant) [inline src/contracts/core/AVSDirectory.sol:845]: make this a 2d array of `[strategies.length][operatorSets.length]`

File: src/contracts/core/AVSDirectory.sol [hunk 3]
```diff
@@ -827,6 +827,42 @@ contract AVSDirectory is
...
+    /*
+    * @param operator the operator to get the slashable magnitude for
+    * @param strategy the strategy to get the slashable magnitude for
+    * @param operatorSet the operatorSet to get the slashable magnitude for
+    * 
+    * @return the current slashable magnitude of the given operator, strategy, and operatorSet
+    */
+    function getCurrentSlashableMagnitude(
+        address operator,
+        IStrategy[] calldata strategy
```
Comment (gpsanant) [inline src/contracts/core/AVSDirectory.sol:839]: strategies

File: src/contracts/core/AVSDirectory.sol [hunk 4]
```diff
@@ -827,6 +827,42 @@ contract AVSDirectory is
...
+    ) public view returns (OperatorSet[] memory, IStrategy[] memory, uint64[] memory) {
+        OperatorSet[] memory operatorSets = operatorSetsMemberOf(operator, 0, inTotalOperatorSets(operator));
+        uint256 maxValues = operatorSets.length * strategy.length;
+        OperatorSet[] memory returnOperatorSets = new OperatorSet[](maxValues);
+        IStrategy[] memory returnStrategies = new IStrategy[](maxValues);
```
Comment (gpsanant) [inline src/contracts/core/AVSDirectory.sol:844]: don't need to keep track of this

File: src/contracts/core/AVSDirectory.sol [hunk 5]
```diff
@@ -827,6 +827,42 @@ contract AVSDirectory is
...
+                slashableMagnitude[index] = _getCurrentMagnitude(operator, strategy[j], operatorSets[i]);
+            }
+        }
+        return (returnOperatorSets, returnStrategies, slashableMagnitude);
+    }
+
+    function _getCurrentMagnitude(address operator, IStrategy strategy, OperatorSet memory operatorSet) internal view returns (uint64) {
+        uint64 currentMagnitude = uint64(
+            // We always do upperLookupLinear here because we want to get the most recent value
+            _magnitudeUpdate[operator][strategy][operatorSet.avs][operatorSet.operatorSetId].upperLookupLinear(uint32(block.timestamp))
```
Comment (gpsanant) [inline src/contracts/core/AVSDirectory.sol:861]: can use `.latest()`

File: src/contracts/core/AVSDirectory.sol (hunk 1 above)
Comment (shrimalmadhur) [reply src/contracts/core/AVSDirectory.sol]: 0ea5394635402ed033a2adb513dbb40d38aab798

File: src/contracts/core/AVSDirectory.sol (hunk 2 above)
Comment (shrimalmadhur) [reply src/contracts/core/AVSDirectory.sol]: 0ea5394635402ed033a2adb513dbb40d38aab798

File: src/contracts/core/AVSDirectory.sol (hunk 3 above)
Comment (shrimalmadhur) [reply src/contracts/core/AVSDirectory.sol]: 0ea5394635402ed033a2adb513dbb40d38aab798

File: src/contracts/core/AVSDirectory.sol (hunk 4 above)
Comment (shrimalmadhur) [reply src/contracts/core/AVSDirectory.sol]: 0ea5394635402ed033a2adb513dbb40d38aab798

File: src/contracts/core/AVSDirectory.sol (hunk 5 above)
Comment (shrimalmadhur) [reply src/contracts/core/AVSDirectory.sol]: 0ea5394635402ed033a2adb513dbb40d38aab798

File: src/contracts/core/AVSDirectory.sol [hunk 6]
```diff
@@ -827,6 +827,30 @@ contract AVSDirectory is
...
+    ) public view returns (OperatorSet[] memory, uint64[][] memory) {
+        OperatorSet[] memory operatorSets = operatorSetsMemberOf(operator, 0, inTotalOperatorSets(operator));
+        uint64[][] memory slashableMagnitudes = new uint64[][](strategies.length);
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            slashableMagnitudes[i] = new uint64[](operatorSets.length);
+            for (uint256 j = 0; j < operatorSets.length; ++j) {
+                uint64 currentMagnitude = uint64(
+                    _magnitudeUpdate[operator][strategies[i]][operatorSets[j].avs][operatorSets[j].operatorSetId].latest()
+                );
+                slashableMagnitudes[i][j] = currentMagnitude;
```
Comment (gpsanant) [inline src/contracts/core/AVSDirectory.sol:848]: one line

**Review by gpsanant — APPROVED** [review]

File: src/contracts/core/AVSDirectory.sol (hunk 6 above)
Comment (shrimalmadhur) [reply src/contracts/core/AVSDirectory.sol]: 6a428a150edd1c18bb839004cdd27f92b7055b0e

**Pull request merged**